helper/temporal/temporal.go

```diff
@@ -0,0 +1,25 @@
+package temporal
+
+import (
+	"gopkg.in/src-d/go-billy.v3"
+	"gopkg.in/src-d/go-billy.v3/util"
+)
+
+// Temporal is a helper that implements billy.TempFile over any filesystem.
+type Temporal struct {
+	billy.Filesystem
+	tmpdir string
+}
+
+// New creates a new filesystem wrapping up 'fs' the intercepts the calls to
+// the TempFile method.
+func New(fs billy.Filesystem, tmpdir string) billy.Filesystem {
+	return &Temporal{
+		Filesystem: fs,
+		tmpdir:     tmpdir,
+	}
+}
+
+func (h *Temporal) TempFile(dir, prefix string) (billy.File, error) {
+	return util.TempFile(h.Filesystem, h.Join(h.tmpdir, dir), prefix)
```

Comment:
According to the TempFile interface description...
> TempFile creates a new temporary file in the directory dir with a name beginning with prefix...
but I believe this method is creating the file in the directory `h.tmpdir/dir`.

Is this correct?